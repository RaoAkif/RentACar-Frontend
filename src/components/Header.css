header {
  width: 100%;
}

.close-arrow,
.hamburger {
  height: 40px;
  width: 40px;
  margin: 2rem;
  cursor: pointer;
  transition: color 0.2s ease-in-out;
  display: block;
}

.hamburger:hover {
  color: var(--green-color);
}

.navigation {
  width: 100vw;
  height: 100vh;
  background-color: #fff;
  position: fixed;
  display: none;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 3000;
  border-right: 1px solid #ededed;
}

.navigation-menu {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.navigation-menu .logo {
  width: 100%;
  display: flex;
  justify-content: center;
  padding: 4rem;
  margin: 2rem 0 6rem 0;
  color: #444;
}

.navigation-menu .logo svg {
  width: 50%;
}

.navigation-menu ul {
  display: flex;
  flex-direction: column;
  width: 100%;
  font-size: 1.8rem;
  font-weight: bold;
  background-color: #fff;
  padding: 0;
}

.navigation-menu li {
  list-style-type: none;
  text-align: center;
  margin: 0;
}

a {
  transition: color 0.1s ease-in-out;
}

a:hover {
  color: #658200 !important;
}

.navigation-menu li a {
  color: #000;
  text-decoration: none;
  display: block;
  width: 100%;
  padding: 1.5rem 0;
}

.navigation-menu li:hover {
  background-color: #eee;
}

.active {
  color: #fff !important;
  background-color: var(--green-color) !important;
}

.nav-footer {
  margin-bottom: 4rem;
  font-size: 1.4rem;
}

.close-btn {
  display: none;
}

.close-btn-mobile {
  color: #ccc;
  font-size: 2rem;
  position: absolute;
  right: 20px;
  top: 20px;
}

@media only screen and (min-width: 480px) {
  .navigation {
    width: 200px;
  }

  .navigation-menu .logo svg {
    width: 100%;
  }

  .navigation-menu ul {
    font-size: 1.6rem;
    padding-left: 2rem;
  }

  .navigation-menu li {
    text-align: left;
  }

  .navigation-menu li a {
    padding-left: 4rem;
  }

  .close-btn {
    width: 7.5rem;
    height: 6rem;
    font-size: 5px;
    position: absolute;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    left: 200px;
    bottom: 40px;
    color: #fff;
    background-color: var(--green-color);
    border-radius: 0 40px 40px 0;
  }

  .close-arrow {
    height: 20px;
    width: 20px;
    margin: 2rem;
    cursor: pointer;
    transition: color 0.2s ease-in-out;
    display: block;
    transform: rotate(-90deg);
    text-shadow: 2px 2px;
  }

  .close-arrow:hover {
    color: #658200 !important;
  }

  .close-btn-mobile {
    display: none;
  }
}
