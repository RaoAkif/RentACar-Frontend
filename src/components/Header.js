import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

const navLinks = {
  listStyle: 'none',
  display: 'flex',
  color: 'azure',
};

function Header() {
  return (
    <div className="header">
      <div className="nav-bar-left">
        <h2 className="logo">MY_LOGO_TEXT</h2>
        <nav>
          <ul style={navLinks}>
            <li>
              <Link className="navLink" to="/">Cake Counter</Link>
            </li>
            <li>
              <Link className="navLink" to="/icecream">Ice cream Counter</Link>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  );
}

export default Header;
