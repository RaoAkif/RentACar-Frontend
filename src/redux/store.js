import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
