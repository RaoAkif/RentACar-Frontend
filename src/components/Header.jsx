import { useState } from 'react';
import './Header.css';
import { HiMenuAlt4 } from 'react-icons/hi';
import Navbar from './Navbar';

function Header() {
  const [transitionState, setTransitionState] = useState(false);
  const transitions = {
    entering: {
      display: 'flex',
    },
    entered: {
      opacity: 1,
      display: 'flex',
    },
    exiting: {
      opacity: 0,
      display: 'flex',
    },
    exited: {
      opacity: 0,
      display: 'none',
    },
  };

  const handleMenu = () => {
    setTransitionState(!transitionState);
  };

  return (
    <>
      <Navbar
        handleMenu={handleMenu}
        transitionState={transitionState}
        transitions={transitions}
      />
      <header>
        <HiMenuAlt4 className="hamburger" onClick={handleMenu} />
      </header>
    </>
  );
}

export default Header;
