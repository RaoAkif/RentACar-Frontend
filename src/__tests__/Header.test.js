import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import Header from '../components/Header';

describe('Header', () => {
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
        ,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
