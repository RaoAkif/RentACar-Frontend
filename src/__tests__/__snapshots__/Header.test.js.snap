// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <nav
    class="navigation"
    style="transition: all .2s; opacity: 0; display: none;"
  >
    <svg
      class="close-btn-mobile"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
      />
    </svg>
    <div
      class="navigation-menu"
    >
      <a
        class="logo"
        href="/"
      >
        <svg>
          rentacar_logo.svg
        </svg>
      </a>
      <p
        class="user-name"
      >
        Signed in as 
        <span />
      </p>
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            CARS
          </a>
        </li>
        <li>
          <a
            class=""
            href="/add_rental"
          >
            NEW RENTAL
          </a>
        </li>
        <li>
          <a
            class=""
            href="/rentals"
          >
            RENTALS
          </a>
        </li>
        <li>
          <a
            class=""
            href="/add_car"
          >
            ADD CAR
          </a>
        </li>
        <li>
          <a
            class=""
            href="/delete_car"
          >
            DELETE CAR
          </a>
        </li>
        <li>
          <a
            class=""
            href="/login"
          >
            LOGIN
          </a>
        </li>
      </ul>
    </div>
    <div
      class="nav-footer"
    >
      © 2022 Microverse
    </div>
    <div
      class="close-btn"
    >
      <svg
        class="close-arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M247 65.16L32.34 440.8l61.79-35.7L247 137.6zm18 .38V137l158.3 271.3 62.7 36.1C412.2 318.2 338.6 191.8 265 65.54zM415.4 424.5l-321.3 1.4-62.72 36.2 445.82-1.9z"
        />
      </svg>
    </div>
  </nav>
  <header>
    <svg
      class="hamburger"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3 7a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 13a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
        fill-rule="evenodd"
      />
    </svg>
  </header>
  ,
</div>
`;
